### PR TITLE
ci: remove jenkins from master branch

### PR DESCRIPTION
**Pull request by leseb**
Thank you Jenkins for all those years, but it's to retire.

Signed-off-by: Sébastien Han <seb@redhat.com>

**Description of your changes:**

**Which issue is resolved by this Pull Request:**
Resolves #

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/master/development-flow.html#commit-structure).
- [ ] **Skip Tests for Docs**: Add the flag for skipping the build if this is only a documentation change. See [here](https://github.com/rook/rook/blob/master/INSTALL.md#skip-ci) for the flag.
- [ ] **Skip Unrelated Tests**: Add a flag to run tests for a specific storage provider. See [test options](https://github.com/rook/rook/blob/master/INSTALL.md#test-storage-provider).
- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/master/development-flow.html#submitting-a-pull-request)
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.
- [ ] Pending release notes updated with breaking and/or notable changes, if necessary.
- [ ] Upgrade from previous release is tested and upgrade user guide is updated, if necessary.
- [ ] Code generation (`make codegen`) has been run to update object specifications, if necessary.

### CODE REVIEW

File: build/release/Jenkinsfile.tag [hunk 1]
```diff
@@ -1,34 +0,0 @@
-// Rook build for Jenkins Pipelines
```
Comment (travisn) [inline build/release/Jenkinsfile.tag:1]: Before we get rid of the tag pipeline, we need a replacement for releases so we make sure to create the release branch and push tags safely. Maybe it's a simple `make release tag` command that the person running the release would run, but we at least need documentation and we need it to be as foolproof as this pipeline was. For example, I don't want to depend on the developer's local enlistment being in a clean state. Thoughts?

**Review by travisn — CHANGES_REQUESTED** [review]

File: build/release/Jenkinsfile.tag (hunk 1 above)
Comment (leseb) [reply build/release/Jenkinsfile.tag:1]: If it's automated with the make, which user should do the tag? I agree we should not depend on the dev env. We just need to think about how we want to trigger the release.
It an issue if this file is removed from master? It's where Jenkins looks at?

Comment (travisn) [reply build/release/Jenkinsfile.tag:1]: Before we remove these, I just want to double check how Jenkins is expected to behave when they're removed from master.

**Review by travisn — APPROVED** [review]
Ok, let's make the conversion!

**Pull request merged**